client/my-sites/plans-features-main/index.tsx

```diff
@@ -437,10 +455,7 @@ const PlansFeaturesMain = ( {
 					isCustomDomainAllowedOnFreePlan={ isCustomDomainAllowedOnFreePlan }
 					onClose={ toggleIsFreePlanPaidDomainDialogOpen }
 					onFreePlanSelected={ () => {
-						if ( isCustomDomainAllowedOnFreePlan.isLoading ) {
-							return;
-						}
-
+						// The modal is only shown if the experiment is already loaded
```

Comment:
It's true but that's not removing this in the name of defensive programming :) 
By doing so this callback function would couple with the external context. Although unlikely, if someone mistakenly remove the loading check and show the modal instantly, there will be a small time window that this dialog can do the wrong thing. 
To be honest the `return` here isn't perfect, but it's better than nothing :)

Reply:
Right I have added this back!